### PR TITLE
New Xenoarch Artifact

### PR DESCRIPTION
**Pull request by clusterfack**
Adds a new xenoarch artifact. Its a communications device that starts with 6 stones, upon taking a stone and using .y as your channel you can communicate with any other person holding a stone regardless of location. All that is required is the actual ancient device be powered. You can place the stones back into the device at any time by attacking it with one.

While I was at it, I modified how inherent channels work. Now all ghosts will have a follow reference for inherent channels (and the (follow) is left justified like it is everywhere else), as all normal radio channels work right now.

### CODE REVIEW

**Comment by PJB3005** [comment]
Xenoarch telepathy?

I like it!

**Comment by clusterfack** [comment]
Alright, this is complete and fully tested, everything works as intended.

**Comment by Arthurdentist** [comment]
Love this, I've long wanted more xenoarch artifacts. Great addition

File: baystation12.dme [hunk 1]
```diff
@@ -482,7 +482,6 @@
 #include "code\game\machinery\embedded_controller\airlock_controllers.dm"
 #include "code\game\machinery\embedded_controller\airlock_program.dm"
 #include "code\game\machinery\embedded_controller\embedded_controller_base.dm"
-#include "code\game\machinery\kitchen\chickenprocessor.dm"
```
Comment (ComicIronic) [inline baystation12.dme]: Why

File: code/game/say.dm [hunk 2]
```diff
@@ -127,18 +127,6 @@ var/list/freqtoname = list(
 		return returntext
 	return "[copytext("[freq]", 1, 4)].[copytext("[freq]", 4, 5)]"
 
-/proc/attach_spans(input, list/spans)
```
Comment (ComicIronic) [inline code/game/say.dm]: WHY

File: code/__HELPERS/unsorted.dm [hunk 3]
```diff
@@ -1337,4 +1337,12 @@ proc/rotate_icon(file, state, step = 1, aa = FALSE)
 
 /proc/isEmag(obj/O)
 	if(!O) return 0
-	return istype(O, /obj/item/weapon/card/emag)
\ No newline at end of file
+	return istype(O, /obj/item/weapon/card/emag)
```
Comment (ComicIronic) [inline code/__HELPERS/unsorted.dm]: Make this a macro

File: code/modules/mob/living/carbon/alien/say.dm [hunk 4]
```diff
@@ -12,10 +12,10 @@
 		return
 
 	var/message_a = say_quote(message)
-	var/rendered = "<i><span class='game say'>Hivemind, <span class='name'>[name]</span> <span class='message'>[message_a]</span></span></i>"
-	for (var/mob/living/S in player_list)
-		if((!S.stat && (S.hivecheck())) || ((S in dead_mob_list) && !istype(S, /mob/new_player)))
-			S << rendered
+	var/rendered = text("<i><span class='game say'>Hivemind, <span class='name'>[]</span> <span class='message'>[]</span></span></i>",name,message_a)
+	for (var/mob/S in player_list)
+		if(S.hivecheck() || ((S in dead_mob_list) && !istype(S, /mob/new_player)))
```
Comment (ComicIronic) [inline code/modules/mob/living/carbon/alien/say.dm]: Why remove the stat check?

File: baystation12.dme (hunk 1 above)
Comment (clusterfack) [inline baystation12.dme]: Whoops, must have loaded the dme improperly after checking out

File: code/game/say.dm (hunk 2 above)
Comment (clusterfack) [inline code/game/say.dm]: Yeah I dont know how that happened

File: code/__HELPERS/unsorted.dm (hunk 3 above)
Comment (clusterfack) [inline code/__HELPERS/unsorted.dm]: You mean the emag proc?

File: code/modules/mob/living/carbon/alien/say.dm (hunk 4 above)
Comment (clusterfack) [inline code/modules/mob/living/carbon/alien/say.dm]: Yeah you're right on that one it should prevent hearing things while in crit

**Comment by clusterfack** [comment]
Alright

**Pull request merged**